.ci/update-github-page.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+set -e
+
+checkForVariable() {
+  VAR_NAME=$1
+  if [ ! -v "$VAR_NAME" ]; then
+    echo "Error: Define $1 environment variable"
+    exit 1
+  fi
+}
+
+checkForVariable "READ_ONLY_TOKEN"
+
+checkout_from https://github.com/checkstyle/contribution
+
+cd .ci-temp/contribution/releasenotes-builder
+mvn -e --no-transfer-progress clean compile package
+cd ../../../
+
+if [ -d .ci-temp/checkstyle ]; then
+  cd .ci-temp/checkstyle/
+  git reset --hard origin/master
+  git pull origin master
+  git fetch --tags
+  cd ../../
+else
+  cd .ci-temp/
+  git clone https://github.com/checkstyle/checkstyle
+  cd ../
+fi
+
+CS_RELEASE_VERSION="$(getCheckstylePomVersion)"
+echo CS_RELEASE_VERSION="$CS_RELEASE_VERSION"
+
+cd .ci-temp/checkstyle
+LATEST_RELEASE_TAG=$(curl -s https://api.github.com/repos/checkstyle/checkstyle/releases/latest \
+                       | jq ".tag_name")
+echo LATEST_RELEASE_TAG="$LATEST_RELEASE_TAG"
+
+cd ../
+
+BUILDER_RESOURCE_DIR="contribution/releasenotes-builder/src/main/resources/com/github/checkstyle"
+
+java -jar contribution/releasenotes-builder/target/releasenotes-builder-1.0-all.jar \
+     -localRepoPath checkstyle \
+     -remoteRepoPath checkstyle/checkstyle \
+     -startRef "$LATEST_RELEASE_TAG" \
+     -releaseNumber "$CS_RELEASE_VERSION" \
+     -githubAuthToken "$READ_ONLY_TOKEN" \
+     -generateGitHub \
+     -gitHubTemplate $BUILDER_RESOURCE_DIR/templates/github_post.template
+
+echo ==============================================
+echo
+echo "GITHUB PAGE:"
+echo ==============================================
+PAGE=$(cat github_post.txt)
+echo "$PAGE"
+
+NEW_RELEASE=$(git describe --abbrev=0 | cut -d '-' -f 2)
+echo NEW_RELEASE="$NEW_RELEASE"
+
+echo "Updating Github tag page"
+curl -i -H "Authorization: token $READ_ONLY_TOKEN" \
+  -d "{ \"tag_name\": \"checkstyle-$NEW_RELEASE\", \
```

Comment:
I believe you can use single quotes and avoid escape for inner double quotes

Reply:
we do not.
`-d "` is required to evaluate `$NEW_RELEASE`.
`\"tag_name\"` is required because json requires `"` and we used `-d "`

I would be happy to find some good workaround. @strkkk , please suggest.

Reply:
I took this as challenge, 

```
$ NEW_RELEASE='10.3.1'
$ echo '{ "tag_name": "checkstyle-'$NEW_RELEASE'",'
{ "tag_name": "checkstyle-10.3.1",
```

@Rahulkhinchi03 , please use:
```

curl -i -X POST https://api.github.com/repos/checkstyle/checkstyle/releases \
  -H "Authorization: token $RAD_ONLY_TOKEN" \
  -d '{ "tag_name": "checkstyle-'$NEW_RELEASE'", \
        "target_commitish": "master", \
        "name": "", \
        "body": "'$PAGE'", \
        "draft": false,   "prerelease": false } \
     '
```

Reply:
I still could not resolve this issue. Please help with the problem parsing JSON error.

```
echo "Updating Github tag page"
curl -i -X POST https://api.github.com/repos/Rahulkhinchi03/checkstyle/releases \
  -H "Authorization: token "$TOKEN"" \
  -d "{ \"tag_name\": \"checkstyle-$NEW_RELEASE\", \
        \"body\": \"$CONTENT\" }"
```

Reply:
> -H "Authorization: token "$TOKEN""

`-H "Authorization: token $TOKEN"`
no need in extra `"`

--------

always place error message that you see.

Reply:
Did that, still an error. The same log as above.

```
echo "Updating Github tag page"
curl -i -X POST https://api.github.com/repos/Rahulkhinchi03/checkstyle/releases \
  -H "Authorization: token $TOKEN" \
  -d "{ \"tag_name\": \"checkstyle-$NEW_RELEASE\", \
        \"body\": \"$CONTENT\" }"
```

Reply:
POST means create new, it is done during release creation when we upload jar.
You do update, please try PATCH method, as you update some fields.
Please read official documentation on how to do update (not a creation)

Reply:
It might be due to multi line, Json does not support multiline.
Remove all `\n` from content of `$CONTENT`  variable.

Reply:
There is no `\n` in the `$CONTENT`

Reply:
Didn't work

Reply:
@Rahulkhinchi03 please paste exact content of `CONTENT` variable here in a comment.

> Didn't work

This is not helpful, how exactly did you try to replace new lines?

Reply:
```
echo "Updating Github tag page"
 curl \
  -X PATCH https://api.github.com/repos/checkstyle/checkstyle/releases/"$RELEASE_ID" \
  -H "Accept: application/vnd.github+json" \
  -H "Authorization: token $GITHUB_TOKEN" \
  -d "{ \"tag_name\": \"checkstyle-$TARGET_RELEASE_NUM\", \
        \"body\": \"$CONTENT\" }"
```

I tried some StackOverflow solutions:
`\"body\": \\\"$CONTENT\\\" }"` and `\"body\": \"$CONTENT/n\" }"` and `\"body\": \"$CONTENT/n/r\" }"`

Reply:
Content of $CONTENT:
```
Checkstyle 10.1 - https://checkstyle.org/releasenotes.html#Release_10.1

Breaking backward compatibility:

  #9146 - upgrade minimal required jdk version to 11
  #10907 - LineLengthCheck ignores length of import and package statements.

New:

  #11259 - RedundantModifier should include RECORD_DEF in acceptable tokens

Bug fixes:

  #10924 - All checks that use `AbstractCheck#getLine()` should check code points instead of characters for spacing
  #11213 - False positive: SummaryJavadocCheck
  #10776 - Inline @return tag does not pass SummaryJavadocCheck
  #11357 - UnnecessaryParentheses reports incorrect violation message for return value in lambdas
  #11030 - Enhance logging approach of SummaryJavadocCheck and use AST-based approach to get content of inline tag.

<details>
<summary>Other Changes:</summary>
<br/>
  Expand XPath IT Regression Testing <br/>
<br/>
  Update Tests to use new 'verifyXxxxxx' method or 'execute' that use inlined config in Input files <br/>
<br/>
  Conver test for SuppressionXpathSingleFilterTest to use inlined config in Input files <br/>
<br/>
  Conver test for SuppressionXpathFilterTest to use inlined config in Input files <br/>
<br/>
  Specify violation messages in input files. <br/>
<br/>
  Update doc for MultipleStringLiterals <br/>
<br/>
  Move methods common in UnusedLocalVariableCheck and FinalClassCheck to util <br/>
<br/>
  Remove suppressions for original verify method or explain in comment <br/>
<br/>
  Migrate to Inline Config Parser for filters <br/>
<br/>
  Test inputs should be completely standalone. <br/>
<br/>
  spelling: Fix double word in TreeWalker <br/>
<br/>
  Checkstyle should force logging of AST in checks that require it <br/>
<br/>
  refactor check for override annotation - add util method <br/>
<br/>
  Fix existing cases of `getFileContents()` usage <br/>
<br/>
  JavadocMetadataScraper fails silently if it can't write the XML files <br/>
<br/>
  AvoidStarImport: update documentation with example for mix of properties  <br/>
<br/>
  doc: mention that excludes can be class names <br/>
</details>
```

Reply:
Please explain how you replaced new lines in `CONTENT`. Did you cat `CONTENT` after you replaced them to confirm that your method works?

Reply:
The above method didn't work. I did this which escaped the newlines but then it was not expected results. There were no spaces left.

```
UPDATE=${CONTENT//$'\n'/\\n}
echo "$UPDATE"
```

Reply:
`awk` works for me:
```
➜  java git:(master) ✗ cat github.txt 

Breaking backward compatibility:

  #9146 - upgrade minimal required jdk version to 11
  #10907 - LineLengthCheck ignores length of import and package statements.

New:

  #11259 - RedundantModifier should include RECORD_DEF in acceptable tokens
...
➜  java git:(master) ✗ awk '{printf "%s\\n", $0}' github.txt 
\nBreaking backward compatibility:\n\n  #9146 - upgrade minimal required jdk version to 11\n  #10907 - LineLengthCheck ignores length of import and package statements.\n\nNew:\n\n  #11259 - RedundantModifier should include RECORD_DEF in acceptable tokens\n\nBug fixes:\n\n  #10924 - All checks that use `AbstractCheck#getLine()` should check code points instead of characters for spacing\n  #11213 - False positive: SummaryJavadocCheck\n  #10776 - Inline @return tag does not pass ...                                        ➜  java git:(master) ✗ 

```

Reply:
I am not sure if @romani agrees to use `awk`.

Reply:
>I am not sure if @romani prefers to use awk.

Let's solve the problem first, right now we do not even have working solution.

Reply:
@nick-mancuso `awk '{printf "%s\\n", $0}' github.txt ` and `${CONTENT//$'\n'/\\n}` gives me the same result.

Reply:
The` command works` but in curl, the same error, `Problem Parsing JSON`

Reply:
do:

`
```
➜  java git:(master) ✗ awk '{printf "%s\\n", $0}' github.txt |  sed "s/\`/'/g" 

\nBreaking backward compatibility:\n\n  #9146 - upgrade minimal required jdk version to 11\n  #10907 - LineLengthCheck ignores length of import and package statements.\n\nNew:\n\n  #11259 - RedundantModifier should include RECORD_DEF in acceptable tokens\n\nBug fixes:\n\n  #10924 - All checks that use 'AbstractCheck#getLine()' should check code points instead of characters for spacing\n  #11213 - False positive: SummaryJavadocCheck\n  #10776 - Inline @return tag does not pass SummaryJavadocCheck\n  #11357 - UnnecessaryParentheses reports incorrect violation message for return value in lambdas\n  #11030 - Enhance logging approach of SummaryJavadocCheck and use AST-based approach to get content of inline tag.\n\n<details>\n<summary>Other Changes:</summary>\n<br/>\n  Expand XPath IT Regression Testing <br/>\n<br/>\n  Update Tests to use new 'verifyXxxxxx' method or 'execute' that use inlined config in Input files <br/>\n<br/>\n  Conver test for SuppressionXpathSingleFilterTest to use inlined config in Input files <br/>\n<br/>\n  Conver test for SuppressionXpathFilterTest to use inlined config in Input files <br/>\n<br/>\n  Specify violation messages in input files. <br/>\n<br/>\n  Update doc for MultipleStringLiterals <br/>\n<br/>\n  Move methods common in UnusedLocalVariableCheck and FinalClassCheck to util <br/>\n<br/>\n  Remove suppressions for original verify method or explain in comment <br/>\n<br/>\n  Migrate to Inline Config Parser for filters <br/>\n<br/>\n  Test inputs should be completely standalone. <br/>\n<br/>\n  spelling: Fix double word in TreeWalker <br/>\n<br/>\n  Checkstyle should force logging of AST in checks that require it <br/>\n<br/>\n  refactor check for override annotation - add util method <br/>\n<br/>\n  Fix existing cases of 'getFileContents()' usage <br/>\n<br/>\n  JavadocMetadataScraper fails silently if it can't write the XML files <br/>\n<br/>\n  AvoidStarImport: update documentation with example for mix of properties  <br/>\n<br/>\n  doc: mention that excludes can be class names <br/>\n</details>\n%   

```
Please try this:
```
JSON=$(cat <<EOF
{ 
    "tag_name": "checkstyle-$TARGET_RELEASE_NUM",
    "body": "${CONTENT}"
}
EOF
)

```
Then do `-d '${JSON}`, and if that doesn't work, try `-d "${JSON}"`

Reply:
I did:
```
JSON=$(cat <<EOF
{
"tag_name": "checkstyle-$TARGET_RELEASE_NUM",
"body": "${CONTENT}"
}
EOF
)

echo "Updating Github tag page"
 curl \
  -X PATCH https://api.github.com/repos/Rahulkhinchi03/checkstyle/releases/"$RELEASE_ID" \
  -H "Accept: application/vnd.github+json" \
  -H "Authorization: token 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" \
  -d '${JSON}'
```

and also,   `-d "${JSON}"`

SAME PROBLEM, `PROBLEM PARSING JSON`

Reply:
add `{}` around TARGET..., also add `cat` of JSON and share here

Reply:
At this point, content of JSON should be valid json, we do not need escapes if we use single `'`. Place content of JSON in some online json validator and debug failure.

Reply:
@nmancus1 @romani  Please label it: 
[ERROR] Issue #9351 does not have breaking compatibility, bug, miscellaneous, new feature, new module label! Please set label at https://github.com/checkstyle/checkstyle/issues/9351

Reply:
Though it is not printing the release number, i.e. `Checkstyle  - https://checkstyle.org/releasenotes.html#Release_`

Reply:
it is not generated by release notes builder
```
GITHUB PAGE:
==============================================
Checkstyle  - https://checkstyle.org/releasenotes.html#Release_
```

so recheck why your template in builder does not generate version.

Reply:
look at https://github.com/checkstyle/checkstyle/pull/12007#discussion_r950763578

Reply:
Fixed Error.